plugins/destination/snowflake/client/migrate.go

```diff
@@ -66,175 +73,283 @@ func (c *Client) MigrateTables(ctx context.Context, msgs message.WriteMigrateTab
 	return g.Wait()
 }
 
-func (c *Client) listTables(ctx context.Context) ([]string, error) {
-	var tables []string
-	rows, err := c.db.QueryContext(ctx, sqlTableList)
-	if err != nil {
-		return nil, err
-	}
-	defer rows.Close()
-	for rows.Next() {
-		var t string
-		if err := rows.Scan(&t); err != nil {
-			return nil, err
+func (c *Client) autoMigrateTable(ctx context.Context, table, oldTable *schema.Table, changes []schema.TableColumnChange) error {
+	tableName := table.Name
+	for _, change := range changes {
+		var err error
+
+		switch {
+		case change.Type == schema.TableColumnChangeTypeMoveToCQOnly:
+			// Drop PKs automatically (and add PK constraint to _cq_id in next step)
+			if len(oldTable.PrimaryKeys()) > 0 {
+				err = c.alterTableDropPK(ctx, tableName)
```

Comment:
We need to do this here because the order of changes was in reverse in my debugging:
changes[0] is `schema.TableColumnChangeTypeMoveToCQOnly`
changes[1] is `schema.TableColumnChangeTypeUpdate` (add PK)
changes[2] is `schema.TableColumnChangeTypeUpdate` (remove current PK)

Obviously adding another PK doesn't work unless you remove the current one.